notary-sora-integration-test/src/integration-test/kotlin/integration/sora/SoraIntegrationTest.kt

```diff
@@ -0,0 +1,159 @@
+        val bobClientId = "$bobClientName@$domain"
+        val keypairBob = ModelCrypto().generateKeypair()
+        val clientBobCredential = IrohaCredential(bobClientId, keypairBob)
+        integrationHelper.createAccount(bobClientName, domain, keypairBob.publicKey())
+
+        integrationHelper.addIrohaAssetTo(aliceClientId, xorAsset, "1334")
+
+        integrationHelper.transferAssetIrohaFromClient(
+            aliceClientId,
+            keypairAlice,
+            aliceClientId,
+            bobClientId,
+            xorAsset,
+            "descr",
+            "1330"
+        )
+
+        assertEquals(
+            "4",
+            integrationHelper.getAccountAssets(clientAliceCredential, aliceClientId).getOrDefault(xorAsset, "0")
+        )
+        assertEquals(
+            "1330",
+            integrationHelper.getAccountAssets(clientBobCredential, bobClientId).getOrDefault(xorAsset, "0")
+        )
+    }
+
+    /**
+     * Sora distribution.
+     *
+     * @given sora account with 35 xor and alice with 0 xor and bob with 0 xor
+     * @when distribute 17 to alice and 18 to bob
+     * @then alice has 17 xor and bob has 18 xor
+     */
+    @Test
+    fun distributeSoraTest() {
+        val aliceClientName = String.getRandomString(9)
+        val aliceClientId = "$aliceClientName@$domain"
+        val keypairAlice = ModelCrypto().generateKeypair()
+        val clientAliceCredential = IrohaCredential(aliceClientId, keypairAlice)
+        integrationHelper.createAccount(aliceClientName, domain, keypairAlice.publicKey())
+
+        val bobClientName = String.getRandomString(9)
+        val bobClientId = "$bobClientName@$domain"
+        val keypairBob = ModelCrypto().generateKeypair()
+        val clientBobCredential = IrohaCredential(bobClientId, keypairBob)
+        integrationHelper.createAccount(bobClientName, domain, keypairBob.publicKey())
+
+        val soraClientId = "sora@sora"
+        val soraKeyPair =
+            ModelUtil.loadKeypair("deploy/iroha/keys/sora@sora.pub", "deploy/iroha/keys/sora@sora.priv").get()
+        integrationHelper.addIrohaAssetTo(soraClientId, xorAsset, "35")
+
+        integrationHelper.transferAssetIrohaFromClient(
+            soraClientId,
+            soraKeyPair,
+            soraClientId,
+            aliceClientId,
+            xorAsset,
+            "descr",
+            "17"
+        )
+
+        integrationHelper.transferAssetIrohaFromClient(
+            soraClientId,
+            soraKeyPair,
+            soraClientId,
+            bobClientId,
+            xorAsset,
+            "descr",
+            "18"
+        )
+
+        assertEquals(
+            "17",
+            integrationHelper.getAccountAssets(clientAliceCredential, aliceClientId).getOrDefault(xorAsset, "0")
+        )
+        assertEquals(
+            "18",
+            integrationHelper.getAccountAssets(clientBobCredential, bobClientId).getOrDefault(xorAsset, "0")
+        )
+    }
+
+}
```

Comment:
General question: do we need such kind of integration tests for the 'xor' token since there is no special behavior in comparison with any other assets and these cases look like Iroha testing

Reply:
I think yes, since we should check Iroha is properly configured and we provide all necessary permissions for Sora integration.